components/script/bluetooth_blacklist.rs

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use regex::Regex;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+use std::string::String;
+
+const EXCLUDE_READS: &'static str = "exclude-reads";
+const EXCLUDE_WRITES: &'static str = "exclude-writes";
+const FILE_PATH: &'static str = "./resources/gatt_blacklist.txt";
+const VALID_UUID_REGEX: &'static str = "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}";
+
+thread_local!(pub static BLUETOOTH_BLACKLIST: RefCell<Option<BluetoothBlacklist>> =
+              RefCell::new(BluetoothBlacklist::init()));
+
+pub struct BluetoothBlacklist(HashMap<String, ValueType>);
+
+#[derive(Eq, PartialEq)]
+pub enum ValueType {
+    Exclude,
+    ExcludeReads,
+    ExcludeWrites,
+}
+
+impl BluetoothBlacklist {
+    pub fn init() -> Option<BluetoothBlacklist> {
+        let mut bt_blist = BluetoothBlacklist(HashMap::new());
+        if !bt_blist.fill_blacklist() {
+            return None;
+        }
+        Some(bt_blist)
+    }
+
+    pub fn is_excluded(&self, uuid: &str) -> bool {
+        self.0.get(uuid).map_or(false, |value| value.eq(&ValueType::Exclude))
+    }
+
+    pub fn is_excluded_from_reads(&self, uuid: &str) -> bool {
+        self.is_excluded(uuid) ||
+        self.0.get(uuid).map_or(false, |value| value.eq(&ValueType::ExcludeReads))
+    }
+
+    pub fn is_excluded_from_writes(&self, uuid: &str) -> bool {
+        self.is_excluded(uuid) ||
+        self.0.get(uuid).map_or(false, |value| value.eq(&ValueType::ExcludeWrites))
+    }
+
+    fn fill_blacklist(&mut self) -> bool {
+        let path = Path::new(FILE_PATH);
+        let mut file = match File::open(&path) {
+            Ok(file) => file,
+            Err(_) => return false,
+        };
+
+        let mut data = String::new();
```

Comment:
We should use a buffered reader, and go through the file line by line. Some examples [here](https://doc.rust-lang.org/std/io/trait.BufRead.html).